container/libcontainer/handler.go

```diff
@@ -71,21 +71,19 @@ func NewHandler(cgroupManager cgroups.Manager, rootFs string, pid int, includedM
 
 // Get cgroup and networking stats of the specified container
 func (h *Handler) GetStats() (*info.ContainerStats, error) {
-	var cgroupStats *cgroups.Stats
-	readCgroupStats := true
+	ignoreStatsError := false
 	if cgroups.IsCgroup2UnifiedMode() {
-		// On cgroup v2 there are no stats at the root cgroup
-		// so check whether it is the root cgroup
+		// On cgroup v2 the root cgroup stats have been introduced in recent kernel versions,
+		// so not all kernel versions have all the data. This means that stat fetching can fail
+		// due to lacking cgroup stat files, but that some data is provided.
 		if h.cgroupManager.Path("") == fs2.UnifiedMountpoint {
-			readCgroupStats = false
+			ignoreStatsError = true
 		}
 	}
-	var err error
-	if readCgroupStats {
-		cgroupStats, err = h.cgroupManager.GetStats()
-		if err != nil {
-			return nil, err
-		}
+
+	cgroupStats, err := h.cgroupManager.GetStats()
+	if err != nil && !ignoreStatsError {
```

Comment:
does it make sense to log something that we're ignored this error? otherwise we're kinda silencing this error

Reply:
Ahh, yeah, that is smart!

Do you think we should use `klog.V(4)` to avoid log spam, or should we always display it?

Reply:
In libcontainer, they already do this for rootless containers: https://github.com/opencontainers/runc/blob/6bae53f5894d9a9811239deb756406d9652843de/libcontainer/cgroups/fs2/fs2.go#L144-L147

Reply:
I see, a `klog.V(4)` sounds reasonable to avoid spam.

Reply:
Thanks! Will fix!

Reply:
Added a `V(4)` now. Still not sure what to write in order to avoid avoid misunderstanding... :thinking: